cmd/thanos/query_frontend.go

```diff
@@ -148,6 +148,10 @@ func registerQueryFrontend(app *extkingpin.App) {
 
 	cmd.Flag("query-frontend.log-failed-queries", "Log failed queries due to any reason").Default("true").BoolVar(&cfg.CortexHandlerConfig.LogFailedQueries)
 
+	cmd.Flag("query-frontend.enable-failed-queries-cache", "Enable cache for failed queries").Default("false").BoolVar(&cfg.CortexHandlerConfig.EnableFailedQueryCache)
```

Comment:
I'd prefer to not introduce too many cmd args, the thanos default behavior is something like 0 means disabled, see examples here: 

https://github.com/databricks/thanos/blob/db_main/cmd/thanos/query.go#L216

```
	storeResponseTimeout := extkingpin.ModelDuration(cmd.Flag("store.response-timeout", "If a Store doesn't send any data in this specified duration then a Store will be ignored and partial data will be returned if it's enabled. 0 disables timeout.").Default("0ms"))
```

Reply:
What's an alternative for such a feature flag? We want to be able to roll out this feature gradually.
A boolean flag is better than "0 means disabled".  There are many other boolean flags.
```
cmd.Flag("query-frontend.compress-responses", "Compress HTTP responses.").
		Default("false").BoolVar(&cfg.CompressResponses)```

Reply:
emm, those feature flags don't have another value associated like size, duration so it had to be a bool IIUC.

Reply:
I saw what you meant now. @pranavmishradatabricks It's a good idea to you can consolidate enable flag into the cache capacity flag so that if the capacity flag is 0, disable the new feature.

Reply:
Consolidated into one flag so that new feature is disabled when capacity is set to 0

Reply:
thanks, please add this note into the help message that 0 means this feature is disabled.